node/node_handler.go

```diff
@@ -601,23 +628,43 @@ func (node *Node) processEpochShardState(epochShardState *types.EpochShardState)
 			utils.GetLogInstance().Info(fmt.Sprintf("[Resharded][epoch:%d] I stay at shard %d, %s", epoch, myShard, aboutLeader), "BlsPubKey", hex.EncodeToString(myBlsPubKey))
 		} else {
 			utils.GetLogInstance().Info(fmt.Sprintf("[Resharded][epoch:%d] I got resharded to shard %d from shard %d, %s", epoch, myShard, node.blockchain.ShardID(), aboutLeader), "BlsPubKey", hex.EncodeToString(myBlsPubKey))
+			node.storeEpochShardState(epochShardState)
 
 			execFile, err := lookPath()
 			if err != nil {
 				utils.GetLogInstance().Crit("Failed to get program path when restarting program", "error", err, "file", execFile)
 			}
-
-			utils.GetLogInstance().Info("Restarting program", "args", os.Args, "env", os.Environ())
-			// TODO: enable restart
-			//err = syscall.Exec(execFile, os.Args, os.Environ())
-			//if err != nil {
-			//	utils.GetLogInstance().Crit("Failed to restart program after resharding" , "error", err)
-			//}
-			//os.Exit(0)
+			args := os.Args
```

Comment:
please wrap the argument setting in a function.

Reply:
Done.